src/metabase/query_processor/pivot/postprocess.clj

```diff
@@ -195,6 +208,10 @@
         (update :row-count (fn [v] (if v (inc v) 0)))
         (update :data update-in (concat row-path col-path)
                 #(update-aggregate (or % (zipmap pivot-measures (repeat {}))) measure-vals measures))
+        (update :row-paths
+                #(when (seq row-path) (assoc-in-path-tree % (butlast row-path) (last row-path))))
+        (update :col-paths
+                #(when (seq col-path) (assoc-in-path-tree % (butlast col-path) (last col-path))))
```

Comment:
`butlast` is usually a bad sign, but I can look at it during the next iteration, doesn't need to be addressed outright.

Reply:
Refactored the code to avoid it